src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java

```diff
@@ -154,12 +164,38 @@ public void hearingResponseUpdate(HearingWrapper wrapper, HmcUpdateResponse resp
             event.getDescription());
 
         log.info("Case Updated with Hearing Response for Case ID {}, Hearing ID {}, Hearing State {} and CCD Event {}",
-            caseData.getCcdCaseId(),
+            caseId,
             hearingRequestId,
             wrapper.getState().getState(),
             event.getEventType().getCcdType());
     }
 
+    @Recover
+    public void hearingResponseUpdateRecover(UpdateCaseException exception, HearingWrapper wrapper, HmcUpdateResponse response) {
+
+        SscsCaseData caseData = wrapper.getCaseData();
+        Long hearingRequestId = response.getHearingRequestId();
+        String caseId = caseData.getCcdCaseId();
+
+        log.info("Updating Case with Hearing Response has failed {} times, sending cancellation request to HMC, for Case ID {}, Hearing ID {} and Hearing State {} with the exception: {}",
+            HEARING_RESPONSE_UPDATE_MAX_RETRIES,
+            caseId,
+            hearingRequestId,
+            wrapper.getState().getState(),
+            exception);
+
+        wrapper.setCancellationReason(OTHER);
+
+        cancelHearing(wrapper);
```

Comment:
Thinking about this further I'm not sure if the cancellation route is the right solution. Isn't it possible for the cancellation to fail? Even if it doesn't, if multiple failures occur after a hearing request is sent there will be quite a few cancellation events in the case history...I'm not sure if that's desirable. The solution to that might be to designate a special cancellation reason which we don't update the case on, or only update the case if a hearing as been booked.
We should speak to the POs about this.